src/assets.rs

```diff
@@ -159,12 +159,30 @@ impl HighlightingAssets {
         file_name: impl AsRef<Path>,
         mapping: &SyntaxMapping,
     ) -> Result<Option<SyntaxReferenceInSet>> {
-        let file_name = file_name.as_ref();
-        Ok(match mapping.get_syntax_for(file_name) {
-            Some(MappingTarget::MapToUnknown) => None,
-            Some(MappingTarget::MapTo(syntax_name)) => self.find_syntax_by_name(syntax_name)?,
-            None => self.get_extension_syntax(file_name.as_os_str())?,
-        })
+        Ok(self.get_syntax_for_path(file_name.as_ref(), mapping).ok())
+    }
+
+    fn get_syntax_for_path(
```

Comment:
Hmm it would probably be a good idea to change the pub API. We already deprecated one method. Need to think a bit more about this…

Reply:
I tried it and liked how it turned out. Let me know what you think about it.